lib/modules/datasource/conan/index.ts

```diff
@@ -87,8 +133,17 @@ export class ConanDatasource extends Datasource {
             if (fromMatch?.groups?.version && fromMatch?.groups?.userChannel) {
               const version = fromMatch.groups.version;
               if (fromMatch.groups.userChannel === userAndChannel) {
+                let newDigest: string | undefined = undefined;
+                if (revision) {
+                  const currentPackageName = `${depName}/${version}${userAndChannel.replace(
+                    '@',
+                    '/'
+                  )}`;
+                  newDigest = await this.getNewDigest(url, currentPackageName);
+                }
```

Comment:
Move to `getDigest` function, you need to add it

Sample
https://github.com/renovatebot/renovate/blob/765a76319be09383f1998eb76745f358db867214/lib/modules/datasource/aws-machine-image/index.ts#L67-L95

Reply:
In the example you mentioned, the digest fetching is kept inside `getReleases`
https://github.com/renovatebot/renovate/blob/765a76319be09383f1998eb76745f358db867214/lib/modules/datasource/aws-machine-image/index.ts#L101-L120
If I do not do it in `getReleases`, the thing I mentioned here happens:  https://github.com/renovatebot/renovate/pull/16871#discussion_r935951273